cmd/metacache-walk.go

```diff
@@ -138,9 +143,6 @@ func (s *xlStorage) WalkDir(ctx context.Context, opts WalkDirOptions, wr io.Writ
 		}
 		dirObjects := make(map[string]struct{})
 		for i, entry := range entries {
-			if len(prefix) > 0 && !strings.HasPrefix(entry, prefix) {
-				continue
-			}
```

Comment:
This was never necessary since `entry` never has `prefixed` paths from the returned values from ListDir